### ISSUE TITLE
Incorrect ArTran.status of 'A' instead of 'T' for financing, such as 'FID','DTD'

### ISSUE THREAD

**Issue by kbower** [kbower/tickettest1#1113]
_Imported from trac ticket 1113.
Created by **kbower**
Opened in trac: Oct 5, 2011, 11:28:05 AM
Last modified in trac: Oct 6, 2011, 9:18:38 AM_
***
_by kbower:_
It seems 'A' is only used for the 'SAL', 'CRM', 'MCR', 'MDB' and 'EEX' transactions (that have an attached order).

**Ivan Smith**
```
SQL> select TRN_TP_CD, STAT_CD, count(*) from ar_trn where STAT_CD != 'P' group by TRN_TP_CD, STAT_CD order by stat_cd
  2  ;


TRN S   COUNT(*)
--- - ----------
CRM A        371
DEP A          1
DTD A          3
EEX A         25
MCR A          2
MDB A          1
SAL A       5929
SER A        781
ADV T          1
ATO T          1
BDR T          9
CON T          1
CRM T         34
DEP T        265
DTC T        125
DTD T        125
EEX T          3
FAC T          2
FCC T          2
FCD T          5
FCW T          2
FDC T         27
FEA T          1
FIC T         56
FID T         64
FOC T          1
IAC T         24
IAD T         36
ICC T         31
ICD T         41
IDC T          1
IDD T          2
LC  T          1
LCC T          5
LEE T          1
MCR T          8
MIS T          1
MSC T          1
MSD T          1
NSF T          3
PMT T        886
R   T         21
REP T          2
SAL T        516
SER T         68
WRC T          8

46 rows selected.
```
**Kane's**
```
SQL> select transactioncodeid, status, count(*) from artrans where status != 'P' group by transactioncodeid, status order by status
  2
SQL> /

TRA S   COUNT(*)
--- - ----------
CRM A        157
EEX A        243
MCR A        105
MDB A         13
SAL A       4054
SER A        781
CRM T        150
DEP T       2981
DTC T        361
DTD T        357
EEX T        144
MCR T        151
MDB T        100
NSF T          1
PMT T       1108
R   T        258
SAL T       3269
SER T        550
WRC T          1
WRD T        416

20 rows selected.
```

**Comment by kbower** [comment]
_**kbower** trac comment on Oct 5, 2011, 11:57:55 AM:_

(Changeset [[1773]](../commit/d420bb69adcfe76614e53459fa4c04f441ed7415))
 * make ArTran._order_based_defaults() default to status 'T' instead of 'A' (see #1113)

**Comment by kbower** [comment]
_**kbower** trac comment on Oct 5, 2011, 1:09:53 PM:_

Nothing hits the customer's account when a reservation is done.
 
GERS has Anticipated, Temporary, and Permanent Transactions (Appear as 'A','T', or 'P')
 
An "Anticipated" Debit is created in A/R at the moment the Sales Order is created.
(This means the order is OPEN and is not yet delivered)
 
This is changed to a "Temporary" Debit the moment the order is Finalized (Delivered) or Voided.
 
Finally, this is converted into a "Permanent" Debit after the day has been balanced and certain reports have been run by the accounting department.
 
This also applies to Credit Memos, Misc. Credit, and Misc. Debits as well.
Payments and Deposits do NOT have "Anticipated" transactions.  They always appear as either "T" or "P".
 
As long as a Sales Order appears as "Anticipated" (Open), you can add and cancel lines.  These changes are reflected in the A/R.
 
Hope this helps.
 
BTW:  June is here, and I promised my people they would have a working demo by now. No presssure!
Jesús R Capó, VP, CIO

**Comment by kbower** [comment]
_**kbower** trac comment on Oct 5, 2011, 1:11:04 PM:_

JR's explanation seems consistent with this:

-------- Original Message --------
Subject:         RE: Ar_trn status
Date:         Wed, 5 Oct 2011 10:30:28 -0400
From:         Wayne Liburd <wliburd@kanesfurniture.com>
To:         Kent Bower <kb@retailarchitects.com>



Kent,

'CRM', 'MCR', 'MDB' and 'EEX' should also be 'A'.

Thanks,

Wayne Liburd
Director of MIS
Kanes Furniture Corporation
5700 70th Ave North
Pinellas Park FL 33781
Tel: +1 (727) 545-9555 ext. 350
Fax: +1 (727) 541-6960

-----Original Message-----
From: Kent Bower [mailto:kb@retailarchitects.com]
Sent: Wednesday, October 05, 2011 10:19 AM
To: Wayne Liburd
Subject: Ar_trn status

Wayne,
     Which AR_TRN's should be written with status = 'A' and which with
'T'?

My understanding is 'SAL' should be 'A', 'DEP' and 'PMT' should be 'T'.

What about the others?

Thanks!
Kent

**Comment by kbower** [comment]
_**kbower** trac comment on Oct 5, 2011, 3:59:41 PM:_

I did a query and notice that CRM, MDB, MCR can be either 'A' or 'T'.  
When are they "A"?
----
Kent,

They are 'A' when the 'SO' transaction is open however, when it is
finalized or voided it is changed to 'T'.

Thanks,
 
Wayne Liburd

**Comment by kbower** [comment]
_**kbower** trac comment on Oct 5, 2011, 5:57:58 PM:_

Kent,

You should not have an MDB or MCR AR_TRN record without a corresponding MDB or MCR SO record.

Wayne

**Comment by kbower** [comment]
_**rneville** trac comment on Oct 6, 2011, 9:18:38 AM:_

(Changeset [[1780]](../commit/23f9380a53a5ff3c585da55254923f7fa1b5b100))
 * fixes #1113 ArTran.status should only be 'A' for debits tied to orders (i.e. SAL, EEX, CRM, MCR, MDB)

**Comment by kbower** [comment]
_**rneville** trac status change on Oct 6, 2011, 9:18:38 AM:_ **closed**
